common/lib/capa/capa/tests/test_html_render.py

```diff
@@ -273,11 +260,39 @@ def test_xml_comments_and_other_odd_things(self):
 
         # Render the HTML
         the_html = problem.get_html()
-        self.assertRegexpMatches(the_html, r"<div>\s+</div>")
+        self.assertRegexpMatches(the_html, r"<div class=\"question\" id=\"1_question_0\" question_index=\"0\">\s+</div>")
 
     def _create_test_file(self, path, content_str):
         test_fp = self.capa_system.filestore.open(path, "w")
         test_fp.write(content_str)
         test_fp.close()
 
         self.addCleanup(lambda: os.remove(test_fp.name))
+
+    def test_existing_xml_compatibility(self):
+        """
+        Verifies that existing problem's XML is converted to new format.
+
+        In new format single are multiple questions should be come inside <question></question>
+        """
+        xml_str = textwrap.dedent("""\
+            <problem>
+            <p>That is the question</p>
+            <multiplechoiceresponse>
+              <choicegroup type="MultipleChoice">
+                <choice correct="false">Alpha <choicehint>A hint</choicehint>
+                </choice>
+                <choice correct="true">Beta</choice>
+              </choicegroup>
+            </multiplechoiceresponse>
+            <demandhint>
+              <hint>question 1 hint 1</hint>
+              <hint>question 1 hint 2</hint>
+            </demandhint>
+            </problem>
+        """)
+
+        # Create the problem
+        problem = new_loncapa_problem(xml_str)
+        childs = [child.tag for child in problem.tree.getchildren()]  # pylint: disable=no-member
+        self.assertEqual(set(childs), set(['question']))
```

Comment:
In general, XML attribute order isn't defined, and it python in particular, attributes are stored in a dict, which means they won't necessarily have a consistent order (which is why `assertXmlEqual` provides more consistent results). As for the question of the new attributes: rather than using `tostring` on the XML tree coming from the problem, you can parse the expected XML, and then compare the XML trees with `assertXmlEqual`.